src/Controls/src/Xaml/MarkupExtensions/OnPlatformExtension.cs

```diff
@@ -107,42 +108,42 @@ MemberInfo minforetriever()
 
 		bool TryGetValueForPlatform(out object value)
 		{
-			if (Device.RuntimePlatform == Device.Android && Android != s_notset)
+			if (DeviceInfo.Platform == DevicePlatform.Android && Android != s_notset)
 			{
 				value = Android;
 				return true;
 			}
-			if (Device.RuntimePlatform == Device.GTK && GTK != s_notset)
+			if (DeviceInfo.Platform == DevicePlatform.Create("GTK") && GTK != s_notset)
```

Comment:
This is probably not worth the code and will be removed in https://github.com/dotnet/maui/pull/4491 but it is here for now and does not break anything.